src/components/DetailsOverlay/Toggle/index.tsx

```diff
@@ -0,0 +1,162 @@
+import { useState } from "react";
+import styled from "@emotion/styled";
+import { palette } from "@leafygreen-ui/palette";
+import Toggle from "@leafygreen-ui/toggle";
+import { Disclaimer } from "@leafygreen-ui/typography";
+import Cookie from "js-cookie";
+import { FORMAT_LOG_V2, PRETTY_PRINT_BOOKMARKS } from "constants/cookies";
+import { QueryParams } from "constants/queryParams";
+import { size } from "constants/tokens";
+import { useQueryParam } from "hooks/useQueryParam";
+import { FilterRow, StyledSubtitle } from "../styles";
+
+const { gray } = palette;
+
+interface BaseToggleProps {
+  ["data-cy"]?: string;
+  leftLabel?: string;
+  rightLabel?: string;
+  label: string;
+  value: boolean;
+  onChange: (checked: boolean) => void;
+}
+
+const BaseToggle: React.FC<BaseToggleProps> = ({
+  "data-cy": dataCy,
+  rightLabel = "ON",
+  leftLabel = "OFF",
+  label,
+  value,
+  onChange,
+}) => (
+  <FilterRow>
+    <StyledSubtitle> {label} </StyledSubtitle>
+    <ToggleWithLabel>
+      <StyledDisclaimer>{leftLabel}</StyledDisclaimer>
+      <Toggle
+        aria-labelledby="base-toggle"
+        checked={value}
+        data-cy={dataCy}
+        onChange={onChange}
+        size="small"
+      />
+      <StyledDisclaimer>{rightLabel}</StyledDisclaimer>
+    </ToggleWithLabel>
+  </FilterRow>
+);
+
+export const WrapToggle: React.FC = () => {
+  const [wrap, setWrap] = useQueryParam(QueryParams.Wrap, false);
+  return (
+    <BaseToggle
+      data-cy="wrap-toggle"
+      label="Wrap"
+      onChange={setWrap}
+      value={wrap}
+    />
+  );
+};
+
+export const CaseSensitiveToggle: React.FC = () => {
+  const [caseSensitive, setCaseSensitive] = useQueryParam(
+    QueryParams.CaseSensitive,
+    false
+  );
+  return (
+    <BaseToggle
+      data-cy="case-sensitive-toggle"
+      label="Case Sensitive"
+      onChange={setCaseSensitive}
+      value={caseSensitive}
+    />
+  );
+};
+
+export const LogV2Toggle: React.FC = () => {
+  const [logV2, setLogV2] = useState(Cookie.get(FORMAT_LOG_V2) === "true");
+  const onChange = (checked: boolean) => {
+    setLogV2(checked);
+    Cookie.set(FORMAT_LOG_V2, checked.toString(), { expires: 365 });
+  };
+
+  return (
+    <BaseToggle
+      data-cy="format-v2-toggle"
+      label="Format LogV2"
+      onChange={onChange}
+      value={logV2}
+    />
+  );
+};
+
+export const FilterLogicToggle: React.FC = () => {
```

Comment:
This almost feels like the right use case for a radio button? It would allow us to remove the labels from the other options. Your call though!

<img width="779" alt="image" src="https://user-images.githubusercontent.com/9298431/190442295-f8946434-5d77-4133-8993-b3eef1c76078.png">

Reply:
I personally am a fan of the toggles over the radio buttons they seem more intuitive.